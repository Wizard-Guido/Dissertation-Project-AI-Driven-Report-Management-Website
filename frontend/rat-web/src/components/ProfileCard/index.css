.card.profile {
    background-color: #f5f6fa;
    padding: 2rem;
    border: 0;
    transition: all 250ms ease-in;
}

.card-body {
    text-align: center;
}

.card.profile:hover {
    transform: translate(0, -5px);
    box-shadow: 0 10px 20px 0 rgba(48, 55, 66, 0.15);
}

.card.profile:hover .positionjob {
    background-color: #74b9ff;
}

.card-img-top.prof {
    border-radius: 50%;
}

p {
    text-align: center;
}

.positionjob {
    border-radius: 1rem;
    transition: all 200ms ease-in;
}