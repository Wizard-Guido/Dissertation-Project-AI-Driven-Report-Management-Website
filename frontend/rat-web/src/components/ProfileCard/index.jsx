import React from 'react'
// import prof from '../../img/prof.jpeg'
import './index.css'


export default function ProfileCard(props) {

    const {last_name, first_name} = props.userInfo;


    return (
        <div className='profilecard'>
            <div className="card profile" style={{width: "18rem"}}>
                {/* <img src={prof} className="card-img-top prof" alt="..." /> */}
                <div className="card-body">
                    <h5 data-testid="userName" className="card-title">{first_name} {last_name}</h5>
                    <p className="card-text">Super Admin<br/>Total RATs: 100</p>
                    <div className="positionjob">Professor</div>
                </div>
            </div>
        </div>
    )
}
