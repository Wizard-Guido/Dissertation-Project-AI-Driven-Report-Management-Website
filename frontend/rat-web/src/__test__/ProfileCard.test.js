/**
 * @jest-environment jsdom
 */

import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ProfileCard from '../components/ProfileCard';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

const props = {
    userInfo: {
        last_name: "Koo",
        first_name: "William"
    }
};

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ProfileCard userInfo = {props.userInfo}/>, div);
});

it("renders button correctly", () => {
    const {getByTestId} = render(<ProfileCard userInfo = {props.userInfo}/>);
    expect(getByTestId("userName")).toHaveTextContent("William Koo");
});

it("matches snapshot", () => {
    const tree = renderer.create(<ProfileCard userInfo = {props.userInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
});