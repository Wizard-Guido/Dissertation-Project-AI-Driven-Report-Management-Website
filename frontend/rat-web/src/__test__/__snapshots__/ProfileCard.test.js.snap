// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="profilecard"
>
  <div
    className="card profile"
    style={
      {
        "width": "18rem",
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
        data-testid="userName"
      >
        William
         
        Koo
      </h5>
      <p
        className="card-text"
      >
        Super Admin
        <br />
        Total RATs: 100
      </p>
      <div
        className="positionjob"
      >
        Professor
      </div>
    </div>
  </div>
</div>
`;
